clients/da-vinci-client/src/main/java/com/linkedin/davinci/client/DaVinciRecordTransformer.java

```diff
@@ -40,33 +40,46 @@ public abstract class DaVinciRecordTransformer<K, V, O> {
    */
   private final boolean storeRecordsInDaVinci;
 
+  /**
+   * The key schema, which is immutable.
+   */
+  private final Schema keySchema;
```

Comment:
This makes sense, and it's probably what we have to deal with (as being able to mutate this runs into conflicting writer problems).  Though I'm trying to think through what kinds of use cases this might disqualify...

I suppose what this implies is that if you have or want state which is managed by DaVinci, then that means the key for that managed state cannot change.  A user still could conceivably do whatever they want with that key elsewhere, but that state would have to be managed by the user and be external to DVRT.  Not 'necessarily' bad, but it is a limitation that maybe we should make clear.